review-evaluation-metrics.md

```diff
@@ -0,0 +1,133 @@
+### Review of Evaluation Metrics
+
+In this summary, I would like to list evaluation metrics of _Music Generation_ papers and their summaries.
+
+1) **Distribution of Pitches**
+   
+   Idea comes from _Folk-RNN_ paper. They compare distribution of pitches of datasets and generated outputs. I think, we can easily implement this metric. _(This is **unary** feature.)_
+
+2) **Distribution of Number of Tokens**
+   
+    Idea comes from _Folk-RNN_ paper. They compare number of tokens in a song (how many token we have until the token which represent end of the sequence) from dataset and generated outputs. In our case, for example, **-1** in _Koma53_ represent end of the sequence. Thus, we can easily implement this metric.
```

Comment:
Yes, I think **-1** means the end. 😕 We can easily add some token to represent end of the piece. But, if we **just** generate short melodic sequences, it will be unnecessary.

Reply:
It's good we clarified what -1 means...

Having said that could you elaborate what a "token" means?

For short sequences, I agree we don't need this step. IMO we should emphasize how long it would take "knowledgable" people to recognize a sequence in makam and justify that we don't need to generate a sequence ending on the final (karar/tonic) note.

Reply:
Token means that _<Koma53(KomaAE), duration>_. End token can be specific like _<-9,0>_

I also got your point. For LSTM system, we can introduce end token, however, for C-RNN-GAN and Midi-Net, most likely, we can not.